libs/x-content/src/main/java/org/opensearch/common/xcontent/XContentType.java

```diff
@@ -199,12 +156,8 @@ public String format() {
         return subtype();
     }
 
-    /** Converts from a {@link MediaType} to an explicit {@link XContentType} */
-    public static XContentType fromMediaType(MediaType mediaType) {
-        if (mediaType instanceof XContentType) {
-            return (XContentType) mediaType;
-        } else {
-            return mediaType != null ? MEDIA_TYPE_PARSER.fromMediaType(mediaType.mediaTypeWithoutParameters()) : null;
-        }
+    @Override
+    public void writeTo(StreamOutput output) throws IOException {
+        output.writeString(this.mediaType());
```

Comment:
Quick Note: this changes the bits over the wire from writing the [Enum index value](https://github.com/opensearch-project/OpenSearch/issues/8797) (which was specific to XContentType only) to write the mime type string (shown here) instead. This does slightly increase the data over the wire (in 20 places here) but gives the flexibility to write custom mime types. We may decide not to keep this functionality, but I'll leave that to a follow up on #8800.

Reply:
After further review on the play. (j/k, digging into PR #8826 for refactoring `contentType` "guessing") I think cutting over to mime type is the way we should go. This starts the path to replacing those deprecated methods with reading the mimetype from the header to support the IETF RFC 6838 specification. What that looks like in the future (e.g., possibly a well vetted library instead of custom implementation) will likely look different than this custom implementation we inherited so I'd like to not get hung up on this too much.